### PR TITLE
Use same error for iteration in <=ES5

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
It's 2022, setting the target to ES2015 or above is now reasonable advice.

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -2443,10 +2443,7 @@
         "category": "Error",
         "code": 2568
     },
-    "Type '{0}' is not an array type or a string type. Use compiler option '--downlevelIteration' to allow iterating of iterators.": {
-        "category": "Error",
-        "code": 2569
-    },
+
```
Comment (jakebailey) [inline src/compiler/diagnosticMessages.json:2446]: Probably don't intend to leave an empty line here?

**Review by jakebailey — APPROVED** [review]

File: src/compiler/diagnosticMessages.json (hunk 1 above)
Comment (DanielRosenwasser) [reply src/compiler/diagnosticMessages.json:2446]: Sometimes I do this so that a person can easily find the next available error code slot.

Comment (jakebailey) [reply src/compiler/diagnosticMessages.json:2446]: AFAIK the documented rule is "largest unused code", so old codes are never reused anyway.

Comment (DanielRosenwasser) [reply src/compiler/diagnosticMessages.json:2446]: > so old codes are never reused anyway

oh sweet summer child

**Pull request merged**